core/src/errors.rs

```diff
@@ -1,5 +1,5 @@
 use core::fmt;
-use winter_utils::string::String;
+use winter_utils::string::*;
```

Comment:
Unconditionally exporting `alloc::string::String` from that module and keeping this import the same didn't solve the issue?

I also think there is little benefit to importing from winter_utils for single imports like this, may as well just import from liballoc.

In any case, overall seems fine, just had the question above before signing off.

Reply:
> Unconditionally exporting alloc::string::String from that module and keeping this import the same didn't solve the issue?

I'm going to use `winterfell` as an example. Here are the changes:

https://github.com/hackaugusto/winterfell/commit/3b768418b89654c959088bfe77b6e6d51b8575d3

The error before and after those changes is the same:

```
error: the item `String` is imported redundantly
   --> utils/core/src/errors.rs:6:5
    |
6   | use crate::string::String;
    |     ^^^^^^^^^^^^^^^^^^^^^
    |
   ::: /Users/hack/.rustup/toolchains/nightly-aarch64-apple-darwin/lib/rustlib/src/rust/library/std/src/prelude/mod.rs:125:13
    |
125 |     pub use super::v1::*;
    |             --------- the item `String` is already defined here
```

Reply:
> I also think there is little benefit to importing from winter_utils for single imports like this, may as well just import from liballoc.

I agree. Just like the prelude, I don't want to change the structure of this here.

I think it is bad practice all these re-exports we are doing over every crate (as an example, these changes requires me to change 5 repos and they are depending on each other). But I don't want to touch that in these PRs.